src/test/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayActionTest.java

```diff
@@ -271,4 +286,51 @@ private static boolean canReplay(WorkflowRun b, String user) {
         });
     }
 
+    @Issue("JENKINS-47339")
+    @Test
+    public void rebuild() throws Exception {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                story.j.jenkins.setSecurityRealm(story.j.createDummySecurityRealm());
+                GlobalMatrixAuthorizationStrategy gmas = new GlobalMatrixAuthorizationStrategy();
+                gmas.add(Jenkins.READ, "dev3");
+                List<Permission> permissions = Run.PERMISSIONS.getPermissions();
+                assertThat(permissions, Matchers.hasItem(Item.BUILD));
+                gmas.add(Item.BUILD, "dev3");
+                gmas.add(Item.READ, "dev3");
+
+                story.j.jenkins.setAuthorizationStrategy(gmas);
+
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition("echo 'script to rebuild'", true));
+                WorkflowRun b1 = story.j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+                story.j.assertLogContains("script to rebuild", b1);
+
+                WorkflowRun b2;
+                { // First time around, verify that UI elements are present and functional.
+                    ReplayAction a = b1.getAction(ReplayAction.class);
+                    assertNotNull(a);
+                    assertFalse(canReplay(b1, "dev3"));
+                    assertTrue(canRebuild(b1, "dev3"));
+                    JenkinsRule.WebClient wc = story.j.createWebClient();
+                    wc.login("dev3");
+                    HtmlPage page = wc.getPage(b1, a.getUrlName());
+                    WebAssert.assertFormNotPresent(page, "config");
+                    HtmlForm form = page.getFormByName("rebuild");
+                    HtmlPage redirect = story.j.submit(form);
+                    assertEquals(p.getAbsoluteUrl(), redirect.getUrl().toString());
+                    story.j.waitUntilNoActivity();
+                    b2 = p.getBuildByNumber(2);
+                    assertNotNull(b2);
+                }
+                story.j.assertLogContains("script to rebuild", story.j.assertBuildStatusSuccess(b2));
+                ReplayCause cause = b2.getCause(ReplayCause.class);
+                assertNotNull(cause);
+                assertEquals(1, cause.getOriginalNumber());
+                assertEquals(b1, cause.getOriginal());
+                assertEquals(b2, cause.getRun());
```

Comment:
Might be interesting to verify that, say, parameters are the same as in the original.